### PR TITLE
ztp: Adds LVM operator source crs for ZTP

### PR DESCRIPTION
**Pull request by alosadagrande**
Adds all the required manifests to install and configure the LVM Operator. Basically the subscription, operatorgroup and namespace (install) and the lvmcluster (configuration).

Signed-off-by: Alberto Losada <alosadag@redhat.com>

cc/ @imiller0

### CODE REVIEW

**Comment by alosadagrande** [comment]
I was able to create a container image with the changes: quay.io/alosadag/ztp-site-generator:lvmo

And applied the following site-policy successfully:

```yaml
apiVersion: ran.openshift.io/v1
kind: PolicyGenTemplate
metadata:
  name: "eric1"
  namespace: "ztp-eric1-policies"
spec:
  bindingRules:
    name: "eric1"
  mcp: "master"
  remediationAction: inform
  sourceFiles:
    - fileName: StorageLVMOSubscriptionNS.yaml
      policyName: "lvmo-policy"
    - fileName: StorageLVMOSubscriptionOperGroup.yaml
      policyName: "lvmo-policy"
    - fileName: StorageLVMOSubscription.yaml
      policyName: "lvmo-policy"
    - fileName: StorageLVMCluster.yaml
      policyName: "lvmo-config"
```

**Comment by imiller0** [comment]
Can you fix the commit message. It needs to start with "ztp: " to pass the CI component test.

File: ztp/source-crs/StorageLVMOSubscription.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: odf-lvm-operator
+  namespace: openshift-storage
+  annotations:
+    ran.openshift.io/ztp-deploy-wave: "2"
+spec:
+  channel: "stable-4.11"
```
Comment (imiller0) [inline ztp/source-crs/StorageLVMOSubscription.yaml:9]: Is there a non-versioned "stable" channel? Other operators are not pinned to a specific channel

File: ztp/source-crs/StorageLVMCluster.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  annotations:
+    ran.openshift.io/ztp-deploy-wave: "10"
+spec:
+  storage:
+    deviceClasses:
+    - name: vg1
+      thinPoolConfig:
+        name: thin-pool-1
+        sizePercent: 90
+        overprovisionRatio: 10
```
Comment (imiller0) [inline ztp/source-crs/StorageLVMCluster.yaml:15]: Array entries are hard to remove, so this entry forces users into using the name "vg1". One option would be to use:
```spec: {}```
That would force users to provide all the content (from storage: down) in their PolicyGenTemplate which seems appropriate for something as variable as storage configuration.

File: ztp/source-crs/StorageLVMOSubscription.yaml (hunk 1 above)
Comment (alosadagrande) [reply ztp/source-crs/StorageLVMOSubscription.yaml:9]: Currently only stable-4.11 and stable-4.10 are available for 4.11 and stable-4.11 for OCP 4.12:

```
oc-mirror list operators --catalog=registry.redhat.io/redhat/redhat-operator-index:v4.11 --package=odf-lvm-operator
NAME              DISPLAY NAME      DEFAULT CHANNEL
odf-lvm-operator  ODF LVM Operator  stable-4.11

PACKAGE           CHANNEL      HEAD
odf-lvm-operator  stable-4.10  odf-lvm-operator.v4.10.8
odf-lvm-operator  stable-4.11  odf-lvm-operator.v4.11.3

# oc-mirror list operators --catalog=registry.redhat.io/redhat/redhat-operator-index:v4.12 --package=odf-lvm-operator
NAME              DISPLAY NAME      DEFAULT CHANNEL
odf-lvm-operator  ODF LVM Operator  stable-4.11

PACKAGE           CHANNEL      HEAD
odf-lvm-operator  stable-4.11  odf-lvm-operator.v4.11.3
```

**Comment by alosadagrande** [comment]
Yeah, my bad, forgot to add the ztp: label to the commit I just included in the title.

File: ztp/source-crs/StorageLVMCluster.yaml (hunk 2 above)
Comment (alosadagrande) [reply ztp/source-crs/StorageLVMCluster.yaml]: Ok. Included a default configuration as a comment in the lvmcluster source-cr.  

So, a default PGT looks like this:

```yaml
apiVersion: ran.openshift.io/v1
kind: PolicyGenTemplate
metadata:
  name: "eric1"
  namespace: "ztp-eric1-policies"
spec:
  bindingRules:
    name: "eric1"
  mcp: "master"
  remediationAction: inform
  sourceFiles:
    - fileName: StorageLVMOSubscriptionNS.yaml
      policyName: "lvmo-policy"
    - fileName: StorageLVMOSubscriptionOperGroup.yaml
      policyName: "lvmo-policy"
    - fileName: StorageLVMOSubscription.yaml
      policyName: "lvmo-policy"
    - fileName: StorageLVMCluster.yaml
      policyName: "lvmo-config"
      spec:
        storage:
          deviceClasses:
          - name: vg1
            thinPoolConfig:
              name: thin-pool-1
              sizePercent: 90
              overprovisionRatio: 10
```

Notice that this config will create a VG with all the available devices except the disk where OCP is install. On top of that, thin-pool LVs will be created

File: ztp/source-crs/StorageLVMOSubscriptionNS.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: openshift-storage
+  labels:
+    openshift.io/cluster-monitoring: "true"
+  annotations:
+    workload.openshift.io/allowed: management
```
Comment (imiller0) [inline ztp/source-crs/StorageLVMOSubscriptionNS.yaml:8]: We need to remove this annotation. The LVMO isn't part of the management partition.

**Review by imiller0 — APPROVED** [review]
/lgtm

**Pull request merged**